### PR TITLE
[3426] Provide fake images for snapshot tests

### PR DESCRIPTION
**Pull request by bychkovdmitry**
### 🎯 Goal

We need to get rid of async image loading for snapshot testing.

### 🛠 Implementation details

Providing images for resources via fake `ImageLoader`. 

Unfortunately, due to our complex Coil configuration, was not able to provide proper behavior for UI Components. We should consider using [MockWebServer](https://github.com/coil-kt/coil/blob/main/coil-compose-base/src/androidTest/java/coil/compose/utils/ImageMockWebServer.kt) for that.

### 🎨 UI Changes

| Compose | UI Components |
| --- | --- |
| io getstream chat android uitests snapshot compose UserAvatarTest_testChannelAvatar | io getstream chat android uitests snapshot uicomponents UserAvatarTest_testChannelAvatar |
| io getstream chat android uitests snapshot compose UserAvatarTest_testUserAvatar | io getstream chat android uitests snapshot uicomponents UserAvatarTest_testUserAvatar |

### 🧪 Testing

1. Execute the command below
```
./gradlew stream-chat-android-ui-uitests:executeScreenshotTests -Precord -Pandroid.testInstrumentationRunnerArguments.filter=io.getstream.chat.android.uitests.util.SnapshotTestFilter
```

2. Check generated screenshots 

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [ ] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests
- [x] Comparison screenshots added for visual changes
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

File: stream-chat-android-ui-uitests/src/androidTest/java/io/getstream/chat/android/uitests/snapshot/compose/UserAvatarTest.kt [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        private val channel = Channel(
+            cid = "messaging:123",
+            members = listOf(
+                Member(user = user1),
+                Member(user = user2),
+                Member(user = user3),
+            )
+        )
+    }
+}
```
Comment (skydoves) [inline stream-chat-android-ui-uitests/src/androidTest/java/io/getstream/chat/android/uitests/snapshot/compose/UserAvatarTest.kt:104]: I'm wondering if we could move these duplicated user modes to the `FakeImageLoader` and reuse it on `uicomponents` and `compose`.

File: stream-chat-android-ui-uitests/src/main/res/layout/view_avatar.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    app:streamUiAvatarBorderWidth="0dp"
+    app:streamUiAvatarOnlineIndicatorEnabled="true"
+    />
+
+    <!--<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"-->
+    <!--xmlns:app="http://schemas.android.com/apk/res-auto"-->
+    <!--android:layout_width="wrap_content"-->
+    <!--android:layout_height="wrap_content"-->
+    <!--    >-->
+    <!--</FrameLayout>-->
```
Comment (skydoves) [inline stream-chat-android-ui-uitests/src/main/res/layout/view_avatar.xml:16]: Maybe can be removed? 😁

File: stream-chat-android-ui-uitests/src/androidTest/java/io/getstream/chat/android/uitests/snapshot/compose/UserAvatarTest.kt [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+    private val context: Context get() = InstrumentationRegistry.getInstrumentation().targetContext
+
+    @get:Rule
+    val composeRule = createComposeRule()
+
+    @Test
+    fun testUserAvatar() {
+        composeRule.setContent {
+            ChatTheme(imageLoader = FakeImageLoader(context)) {
```
Comment (skydoves) [inline stream-chat-android-ui-uitests/src/androidTest/java/io/getstream/chat/android/uitests/snapshot/compose/UserAvatarTest.kt:47]: Do we use the new `imageLoaderFactory` after merging [Don't expose Coil ImageLoader in ChatTheme #3441](https://github.com/GetStream/stream-chat-android/pull/3441/files)?

**Review by skydoves — COMMENTED** [review]
Overall looks good to me! I left some minor comments :)

### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [x] New feature tested and works

File: stream-chat-android-ui-uitests/src/androidTest/java/io/getstream/chat/android/uitests/snapshot/compose/UserAvatarTest.kt [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+import com.karumi.shot.ScreenshotTest
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.client.models.Member
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.compose.ui.components.avatar.ChannelAvatar
+import io.getstream.chat.android.compose.ui.components.avatar.UserAvatar
+import io.getstream.chat.android.compose.ui.theme.ChatTheme
+import io.getstream.chat.android.uitests.util.FakeImageLoader
+import org.junit.Rule
+import org.junit.Test
```
Comment (kanat) [inline stream-chat-android-ui-uitests/src/androidTest/java/io/getstream/chat/android/uitests/snapshot/compose/UserAvatarTest.kt:35]: Hey @bychkovdmitry 👋

Shouldn't we use `JUnit5` for the new Test cases?  :)

Comment (bychkovdmitry) [reply stream-chat-android-ui-uitests/src/androidTest/java/io/getstream/chat/android/uitests/snapshot/compose/UserAvatarTest.kt:35]: There's no official support for Junit5 in instrumentation tests. Only this library: https://github.com/mannodermaus/android-junit5. Don't know if it worth to use it extensively as they are always catching up reimplementing junut4 rules (e.g. espresso rules). 
Also: https://issuetracker.google.com/issues/127100532

File: stream-chat-android-ui-uitests/src/androidTest/java/io/getstream/chat/android/uitests/snapshot/compose/UserAvatarTest.kt (hunk 1 above)
Comment (bychkovdmitry) [reply stream-chat-android-ui-uitests/src/androidTest/java/io/getstream/chat/android/uitests/snapshot/compose/UserAvatarTest.kt]: Removed the duplication here

**Review by skydoves — APPROVED** [review]

File: stream-chat-android-ui-uitests/src/androidTest/java/io/getstream/chat/android/uitests/snapshot/compose/UserAvatarTest.kt (hunk 4 above)
Comment (bychkovdmitry) [reply stream-chat-android-ui-uitests/src/androidTest/java/io/getstream/chat/android/uitests/snapshot/compose/UserAvatarTest.kt:35]: Let's do Junit4 instrumentation tests for now and check later if this library can work for us.

**Review by leandroBorgesFerreira — APPROVED** [review]

**Review by TGazica — APPROVED** [review]
Looks good :)

**Pull request merged**